pkg/solana/logpoller/parser.go

```diff
@@ -396,19 +400,26 @@ func (f *addressFilter) Accept(visitor primitives.Visitor) {
 	}
 }
 
-type eventSigFilter struct {
-	eventSig []byte
+type eventSubkeyFilter struct {
+	Subkey         []string
+	ValueComparers []primitives.ValueComparator
 }
 
-func NewEventSigFilter(sig []byte) query.Expression {
-	return query.Expression{
-		Primitive: &eventSigFilter{eventSig: sig},
-	}
+func NewEventSubkeyFilter(subkey []string, valueComparers []primitives.ValueComparator) query.Expression {
+	return query.Expression{Primitive: &eventSubkeyFilter{
+		Subkey:         subkey,
+		ValueComparers: valueComparers,
+	}}
 }
 
-func (f *eventSigFilter) Accept(visitor primitives.Visitor) {
+func (f *eventSubkeyFilter) Accept(visitor primitives.Visitor) {
 	switch v := visitor.(type) {
 	case *pgDSLParser:
-		v.VisitEventSigFilter(f)
+		v.VisitEventSubkeyFilter(f)
 	}
 }
```

Comment:
We can't combine the `eventSigFilter` and `eventSubkeyFIlter`.  Unlike in evm where the event sig is just "topic 0" the event sig in Solana has nothing to do with the event subkeys.  They are two separate columns in the db table, one has to compare the `event_sig` column and the other references an index into the `subkey_values` column.

Also the event sigs have a specific format, it must always be an 8 byte discriminator--we should validate that if we can I guess.  Whereas the subkey values are fields in a struct, and can be any primitive type; these have to be converted with `NewIndexedValue` before storing in the appropriate slot of `subkey_values`

Reply:
Ahhh. I messed that one up. I don't know what I was thinking. So essentially we need both: the event sig filter when subkey filters are provided.

Reply:
Looking at evm again, event sig is always provided. I'll make that change.

Reply:
Just backing out these changes to parser.go should be fine, as long as you're okay with merging my changes first (which add `VisitEventSubkeyFilter` while keeping `VisitEventSigFilter`)